### PR TITLE
14429 Filing Survey dialog (incremental)

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#14429

*Description of changes:*
- added Filing Survey dialog
- renamed class "genErr" -> "font-14"
- added unit test
- app version = 4.5.6
- still WIP but safe to deploy

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -41,6 +41,13 @@
       @retry="fetchData()"
     />
 
+    <FilingSurveyDialog
+      attach="#app"
+      :dialog="filingSurveyDialog"
+      @no="filingSurveyDialog = false"
+      @yes="filingSurveyDialog = false"
```
Comment (severinbeauvais) [inline src/App.vue:48]: Still WIP, but safe.

File: src/App.vue [hunk 2]
```diff
@@ -554,6 +563,10 @@ export default class App extends Vue {
     // reset errors in case this method is invoked more than once (ie, retry)
     this.resetFlags()
 
+    // *** TODO: read cookie; if it doesn't exist or is expired then show survey
+    // for now do this temporarily
+    if (this.$route.hash.includes('#show-survey')) this.filingSurveyDialog = true
```
Comment (severinbeauvais) [inline src/App.vue:568]: Still WIP, but safe.

File: src/dialogs/FilingSurveyDialog.vue [hunk 3]
```diff
@@ -0,0 +1,92 @@
+<template>
```
Comment (severinbeauvais) [inline src/dialogs/FilingSurveyDialog.vue:1]: Example:

image

File: src/dialogs/FilingSurveyDialog.vue [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+          class="mt-6"
+          @change="updateCookie($event)"
+        />
+      </v-card-text>
+
+      <v-card-actions class="d-flex justify-center">
+        <v-btn
+          id="dialog-no-button"
+          large outlined color="primary"
+          class="mr-sm-3 px-8"
```
Comment (severinbeauvais) [inline src/dialogs/FilingSurveyDialog.vue:36]: A bit of right margin to space this button from the one on the right on screens larger than "sm". On narrower screens, this margin is not applied and the buttons are closer together, ie:

image

File: src/dialogs/FilingSurveyDialog.vue [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  @Prop({ default: '' }) readonly attach!: string
+
+  // Pass click events to parent.
+  @Emit() protected no (): void {}
+  @Emit() protected yes (): void {}
+
+  /** Saves a cookie to prevent showing this dialog for a while. */
+  protected updateCookie (val: boolean) {
+    // *** TODO: save a cookie with an expiry date
+    // and maybe delete cookie if not needed?
```
Comment (severinbeauvais) [inline src/dialogs/FilingSurveyDialog.vue:73]: Still WIP, but safe.

File: src/dialogs/FilingSurveyDialog.vue [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+@import '@/assets/styles/theme.scss';
+
+.v-dialog {
+  .v-card {
+    .v-card__title {
```
Comment (severinbeauvais) [inline src/dialogs/FilingSurveyDialog.vue:83]: You might think that I could get away without specifying the parent elements, but actually they are needed to make this CSS _as specific as_ some of the global (base) CSS that we have. Remember, CSS has [specificity rules](https://www.w3schools.com/css/css_specificity.asp).

**Review by JazzarKarim — APPROVED** [review]

**Review by ketaki-deodhar — APPROVED** [review]

**Pull request merged**